### PR TITLE
Add `schedule` command to schedule meetings

### PR DESCRIPTION
**Pull request by Dethada**
Addresses #23 

TODOs
- [x] Write tests for newly added classes
- [x] Change to follow our feature spec

### CODE REVIEW

File: src/test/java/seedu/findvisor/model/person/MeetingTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(null, null));
+        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(START, null));
+        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(null, END));
+
+        // invalid phone numbers
+        assertFalse(Meeting.isValidDateTime(END, START));
+
+        // valid phone numbers
+        assertTrue(Meeting.isValidDateTime(START, END));
+    }
```
Comment (KohGuanZeh) [inline src/test/java/seedu/findvisor/model/person/MeetingTest.java:39]:
```suggestion
    public void constructor_invalidMeeting_throwsIllegalArgumentException() {
        assertThrows(IllegalArgumentException.class, () -> new Meeting(END, START));
    }

    @Test
    public void isValidMeeting() {
        // null meeting
        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(null, null));
        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(START, null));
        assertThrows(NullPointerException.class, () -> Meeting.isValidDateTime(null, END));

        // invalid meeting
        assertFalse(Meeting.isValidDateTime(END, START));

        // valid meeting
        assertTrue(Meeting.isValidDateTime(START, END));
    }
```
I think we should change the name of the test functions to be representative of what they are testing as well as the comments?

**Review by KohGuanZeh — COMMENTED** [review]
Only noticed a minor mistake with the one of the test code. The other parts of the code LGTM!

File: src/test/java/seedu/findvisor/model/person/MeetingTest.java (hunk 1 above)
Comment (Dethada) [reply src/test/java/seedu/findvisor/model/person/MeetingTest.java:39]: Thanks, nice catch!

**Review by KohGuanZeh — APPROVED** [review]

File: src/main/java/seedu/findvisor/logic/commands/ScheduleCommand.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    public static final String MESSAGE_CANNOT_SCHEDULE_MULTIPLE_MEETINGS = "Cannot schedule more than 1"
+            + " meeting with a contact!";
+    public static final String MESSAGE_CANNOT_SCHEDULE_MEETING_IN_THE_PAST = "Cannot schedule meeting that"
+            + " starts before the current time!";
+
+    private final Index targetIndex;
+    private final Meeting meeting;
+
+    /**
+     * Creates an ScheduleCommand to schedule a meetign with the specified {@code Person}
```
Comment (Javiery3889) [inline src/main/java/seedu/findvisor/logic/commands/ScheduleCommand.java:51]: Small typo here.
```suggestion
     * Creates an ScheduleCommand to schedule a meeting with the specified {@code Person}
```

**Review by Javiery3889 — COMMENTED** [review]
Just a small typo found, code LGTM otherwise 👍

**Review by Javiery3889 — APPROVED** [review]

**Review by KohGuanZeh — APPROVED** [review]
LGTM

**Pull request merged**